src/org/elixir_lang/sdk/ElixirSdkType.java

```
@@ -0,0 +1,332 @@
+    ElixirSdkRelease byModuleSdk = getRelease(module != null ? ModuleRootManager.getInstance(module).getSdk() : null);
+
+    return byModuleSdk != null ? byModuleSdk : getRelease(element.getProject());
+  }
+
+  @Nullable
+  public static ElixirSdkRelease getRelease(@NotNull Project project){
+    if(ElixirSystemUtil.isSmallIde()){
+      return getReleaseForSmallIde(project);
+    }
+    return getRelease(ProjectRootManager.getInstance(project).getProjectSdk());
+  }
+
+
+  @NotNull
+  private static String getDefaultSdkName(@NotNull String sdkHome, @Nullable ElixirSdkRelease version){
+    return version != null ? "Elixir " + version.getRelease() : "Unknown Elixir version at " + sdkHome ;
+  }
+
+  @Nullable
+  private ElixirSdkRelease detectSdkVersion(@NotNull String sdkHome){
+    ElixirSdkRelease cachedRelease = mySdkHomeToReleaseCache.get(getVersionCacheKey(sdkHome));
+    if(cachedRelease != null){
+      return cachedRelease;
+    }
+
+    assert !ApplicationManager.getApplication().isUnitTestMode() : "Unit tests should have their SDK versions pre-cached!";
+
+    File elixir = JpsElixirSdkType.getScriptInterpreterExecutable(sdkHome);
+    if(!elixir.canExecute()){
+      String reason = elixir.getPath() + (elixir.exists() ? " is not executable." : " is missing.");
+      LOG.warn("Can't detect Elixir version: " + reason);
+      return null;
+    }
+
+    try{
+      ProcessOutput output = ElixirSystemUtil.getProcessOutput(sdkHome, elixir.getAbsolutePath(), "-v");
+      List<String> lines = output.getExitCode() != 0 || output.isTimeout() || output.isCancelled() ?
+          ContainerUtil.<String>emptyList() : output.getStdoutLines();
+
+      if(lines.size() == 1){
+        ElixirSdkRelease release = ElixirSdkRelease.fromString(lines.get(0));
+        mySdkHomeToReleaseCache.put(getVersionCacheKey(sdkHome), release);
+        return release;
+      }
+    }catch (ExecutionException ignore){
+      LOG.warn(ignore);
```

You could use this idiom of logging a warning for the ignore exceptions in other files where the ignored exception aren't being logged now.